build/templates/grpc_session_options.rst.mako

```diff
@@ -0,0 +1,101 @@
+<%
+    import build.helper as helper
+
+    config = template_parameters['metadata'].config
+    module_name = config['module_name']
+    driver_name = config['driver_name']
+    enums = config['enums']
+%>\
+${helper.get_rst_header_snippet('gRPC Support', '=')}
+
+Support for using ${driver_name} over gRPC
+
+.. py:currentmodule:: ${module_name}
+
+
+Enums
+=====
+
+
+SessionInitializationBehavior
+-----------------------------
+
+.. py:class:: SessionInitializationBehavior
+
+    .. py:attribute:: SessionInitializationBehavior.AUTO
+
+
+        The NI gRPC Device Server will attach to an existing session with the specified name if it exists,
+        otherwise the server will initialize a new session.
+
+        .. note:: When using the Session as a context manager and the context exits, the behavior depends on what happened when the constructor
+            was called. If it resulted in a new session being initialized on the NI gRPC Device Server, then it will automatically close the
+            server session. If it instead attached to an existing session, then it will detach from the server session and leave it open.
```

Comment:
Can we find a way not to replicate all of this information?

Reply:
Chatted separately about this. It seems anything that doesn't come from metadata (that is, anything generated here in nimi-python itself) has some data replication. So I'm going to close this issue for now, but I agree it would be nice to improve on this in the future.